packages/salesforcedx-vscode-soql/src/commands/soqlFileCreate.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { EDITOR_VIEW_TYPE } from '../constants';
+import { telemetryService } from '../telemetry';
+
+export async function soqlOpenNew() {
+  telemetryService.sendCommandEvent('soql_builder_open_new', process.hrtime());
+
+  if (vscode.workspace) {
+    // create untitled file
+    const newFileUri = vscode.Uri.file(
+      path.join(vscode.workspace.rootPath!, 'untitled.soql')
+    ).with({ scheme: 'untitled' });
+    await vscode.workspace.openTextDocument(newFileUri);
```

Comment:
Creating a new soql file feels like a similar UX to creating a new metadata template (create apex class, create lwc, etc). In that flow we prompt for a filename with an input box. If y'all don't think it makes sense here that's cool, just putting the thought out there for discussion.

Reply:
+1 to prompting for a file name

Reply:
Good point. I think there are arguments to be made for both flows.  

Personally, I feel that since .soql files are ephemeral artifacts (they don't actually get deployed anywhere), the persistence of the file is not as important. The 'untitled' user flow allows a user to create and test their query, copy-paste to Apex if necessary, and then save as desired. 

@clairebianchi Which flow do you prefer for SOQL files?

Reply:
Keep in mind that at this point when they save the query results it will use the title of the document, so they will be saving `untitled.soql.csv`. In this case the persistence of the query may not be as important as the persistence of the results but that will still depend on the file being named.

Reply:
I think there is work in the backlog or at least discussion to prompt the user for a filename and location when they save the data. If that work is done and they are promoted with a save dialog then we may not have to be so concerned with asking on the front end with this command.

Reply:
What would happen if I am building more than one query at a time and I launch SOQL builder more than once simultaneously? I think if we go the 'untitled' file route then we need to have some logic to handle that case. VS Code file name format for untitled is "Untitled-1", "Untitled-2" etc. so we need to check if an untitled file name already exists before generating it.

If we let the user pick/name their file then we may not have to worry about this case.

Reply:
This was not an issue for me during testing. The files incremented as you'd expect, VS Code handled it for us.